streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/TaskMovementTest.java

```diff
@@ -96,34 +139,46 @@ public void shouldReturnEmptyMovementsWhenPassedEmptyTaskAssignments() {
             mkEntry(UUID_1, emptyList()),
             mkEntry(UUID_2, emptyList())
         );
-        assertTrue(getMovements(stateConstrainedAssignment, balancedAssignment, maxWarmupReplicas).isEmpty());
+        assertTrue(getMovements(stateConstrainedAssignment, balancedAssignment, emptyMap(), maxWarmupReplicas).isEmpty());
     }
 
     @Test
     public void shouldReturnEmptyMovementsWhenPassedIdenticalTaskAssignments() {
         final int maxWarmupReplicas = 2;
         final Map<UUID, List<TaskId>> stateConstrainedAssignment = mkMap(
-            mkEntry(UUID_1, asList(TASK_0_0, TASK_1_0)),
-            mkEntry(UUID_2, asList(TASK_0_1, TASK_1_1))
+            mkEntry(UUID_1, mkTaskList(TASK_0_0, TASK_1_0)),
+            mkEntry(UUID_2, mkTaskList(TASK_0_1, TASK_1_1))
         );
         final Map<UUID, List<TaskId>> balancedAssignment = mkMap(
-            mkEntry(UUID_1, asList(TASK_0_0, TASK_1_0)),
-            mkEntry(UUID_2, asList(TASK_0_1, TASK_1_1))
+            mkEntry(UUID_1, mkTaskList(TASK_0_0, TASK_1_0)),
+            mkEntry(UUID_2, mkTaskList(TASK_0_1, TASK_1_1))
         );
-        assertTrue(getMovements(stateConstrainedAssignment, balancedAssignment, maxWarmupReplicas).isEmpty());
+        assertTrue(getMovements(stateConstrainedAssignment, balancedAssignment, emptyMap(), maxWarmupReplicas).isEmpty());
     }
 
     @Test
     public void shouldThrowIllegalStateExceptionIfAssignmentsAreOfDifferentSize() {
         final int maxWarmupReplicas = 2;
 
         final Map<UUID, List<TaskId>> stateConstrainedAssignment = mkMap(
-            mkEntry(UUID_1, asList(TASK_0_0, TASK_0_1))
+            mkEntry(UUID_1, mkTaskList(TASK_0_0, TASK_0_1))
         );
         final Map<UUID, List<TaskId>> balancedAssignment = mkMap(
-            mkEntry(UUID_1, asList(TASK_0_0, TASK_1_0)),
-            mkEntry(UUID_2, asList(TASK_0_1, TASK_1_1))
+            mkEntry(UUID_1, mkTaskList(TASK_0_0, TASK_1_0)),
+            mkEntry(UUID_2, mkTaskList(TASK_0_1, TASK_1_1))
         );
-        assertThrows(IllegalStateException.class, () -> getMovements(stateConstrainedAssignment, balancedAssignment, maxWarmupReplicas));
+        assertThrows(IllegalStateException.class, () -> getMovements(stateConstrainedAssignment, balancedAssignment, emptyMap(), maxWarmupReplicas));
+    }
+
```

Comment:
req: Please add a test that verifies the `IllegalStateException` when a task does not have a destination.